lib/async/http/protocol/http1/server.rb

```diff
@@ -91,10 +91,10 @@ def each(task: Task.current)
 								else
 									head = request.head?
```

Comment:
To avoid GC issues, we need to capture the version variable here into a local before setting request to nil.